src/main/java/strategy/CalculateStrategy.java

```diff
@@ -0,0 +1,33 @@
+package strategy;
+
+import db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+import model.Operation;
+
+public class CalculateStrategy {
+    private final Storage storage;
+    private FruitTransaction fruitTransaction;
+    private final Map<Operation, FruitOperation> operationMap;
+
+    public CalculateStrategy(Storage storage) {
+        this.storage = storage;
+        this.fruitTransaction = fruitTransaction;
+
+        operationMap = Map.of(
+                Operation.BALANCE, new BalanceOperation(fruitTransaction, storage),
+                Operation.SUPPLY, new SupplyReturnOperation(fruitTransaction, storage),
+                Operation.RETURN, new SupplyReturnOperation(fruitTransaction, storage),
+                Operation.PURCHASE, new PurchaseOperation(fruitTransaction, storage)
+        );
+    }
```

Comment:
The code should not be implementation-dependent
```suggestion
    public CalculateStrategy(Storage storage,
    Map<Operation, FruitOperation> operationMap) {
        this.storage = storage;
        this.operationMap = operationMap;
    }
```